### ISSUE TITLE
index 104 out of string (IndexError)

### ISSUE THREAD

**Issue by HenraL** [piotrmurach/tty-markdown-cli#1]
### Describe the problem

While attempting to view the markdown from this repository: <https://github.com/paulbouwer/hello-kubernetes/blob/main/docs/deploy-using-helm.md>
I got this error :

```bash
/var/lib/gems/3.1.0/gems/strings-0.2.1/lib/strings/wrap.rb:158:in `insert': index 104 out of string (IndexError)
        from /var/lib/gems/3.1.0/gems/strings-0.2.1/lib/strings/wrap.rb:158:in `block in insert_ansi'
        from /var/lib/gems/3.1.0/gems/strings-0.2.1/lib/strings/wrap.rb:142:in `reverse_each'
        from /var/lib/gems/3.1.0/gems/strings-0.2.1/lib/strings/wrap.rb:142:in `insert_ansi'
        from /var/lib/gems/3.1.0/gems/strings-0.2.1/lib/strings/wrap.rb:111:in `format_line'
        from /var/lib/gems/3.1.0/gems/strings-0.2.1/lib/strings/wrap.rb:30:in `block in wrap'
        from /var/lib/gems/3.1.0/gems/strings-0.2.1/lib/strings/wrap.rb:29:in `map'
        from /var/lib/gems/3.1.0/gems/strings-0.2.1/lib/strings/wrap.rb:29:in `wrap'
        from /var/lib/gems/3.1.0/gems/strings-0.2.1/lib/strings.rb:109:in `wrap'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:441:in `block in max_row_height'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:440:in `map'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:440:in `with_index'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:440:in `max_row_height'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:430:in `block in max_row_heights'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:429:in `each'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:429:in `reduce'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:429:in `max_row_heights'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:359:in `convert_table'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:37:in `convert'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:56:in `block in inner'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:50:in `each'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:50:in `each_with_index'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:50:in `inner'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:70:in `convert_root'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown/converter.rb:37:in `convert'
        from /var/lib/gems/3.1.0/gems/kramdown-2.4.0/lib/kramdown/converter/base.rb:107:in `convert'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown.rb:144:in `parse'
        from /var/lib/gems/3.1.0/gems/tty-markdown-0.7.2/lib/tty/markdown.rb:163:in `parse_file'
        from /var/lib/gems/3.1.0/gems/tty-markdown-cli-0.1.0/exe/tty-markdown:58:in `<top (required)>'
        from /usr/local/bin/tty-markdown:25:in `load'
        from /usr/local/bin/tty-markdown:25:in `<main>'
```

### Steps to reproduce the problem

I am running Raspberry Pi OS Lite (64-bit) on a raspberry pi 4 with 4Gb of ram
I tried running it via the pipe and via argument but the error was the same.
I installed ruby on the system
Then run the `gem install tty-markdown-cli` to install the program

### Actual behaviour

The program crashes

### Expected behaviour

A coloured and parsed markdown

### Describe your environment

* OS version: Raspberry Pi OS Lite (64-bit)
* Ruby version: ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [aarch64-linux-gnu]
* TTY::Markdown CLI version: 0.1.0 (tty-markdown 0.7.2)

**Comment by piotrmurach** [comment]
Hi @HenraL 👋 

Thanks for using the `tty-markdown-cli` gem and reporting this issue.

I can confirm the error. It only happens at certain widths though.

To display the content without error you can specify the terminal width:

```bash
ttymarkdown -w 180 docs/deploy-using-helm.md
```

It would be good to find the actual markdown line that causes the error. I managed to zero in on the culprit being somewhere inside the table. I'd appreciate it if you could investigate further and provide a minimal example of were the wrapping breaks. Ideally, a string that can be replicated with the `Strings.wrap(text)` call.

**Comment by HenraL** [comment]
Dear @piotrmurach ,
Unfortunatly, i tried to see if I could provide more help, but maybe add a debug flag so that you could get more details.
I am afraid I have provided what I could as a non  ruby programmer.
I think I remember even providing the file that I was using when the error occured
I hope that knowing the terminal was over an ssh connection (the main way used to connect to raspberry pies that are meant to be used as servers) and that the terminal was half the size (on the height) of an average monitor screen could help you.
I wish you a good luck in your debugging, I can always re-run or run updated versions of your program if you need (I am in an englosh time-zone)

PS: the link in the issue brings you directly to the file that was problematic

**Comment by piotrmurach** [comment]
Thanks for the replay. I have used the provided file in my original response. That's why I showed you an example of how to run it so that you don't get the error and demonstrate how to view any markdown documents when you find failures. I hoped you would have time to find out which line causes the issue as I'm struggling for time. I need to prioritise my limited time. Also, the string wrapping functionality will be rewritten from scratch at some point in future.
